pkg/manifest/storages/file_storage.go

```diff
@@ -0,0 +1,94 @@
+package storages
+
+import (
+	"encoding/gob"
+	"os"
+	"time"
+
+	"github.com/ddev/ddev/pkg/manifest/types"
+)
+
+func NewFileStorage(fileName string) types.ManifestStorage {
+	return &fileStorage{
+		fileName: fileName,
+		loaded:   false,
+	}
+}
+
+type fileStorage struct {
+	fileName string
+	loaded   bool
+	data     fileStorageData
+}
+
+// fileStorageData is the structure used for the file.
+type fileStorageData struct {
+	LastUpdate time.Time
+	Manifest   types.Manifest
+}
+
+func (s *fileStorage) LastUpdate() time.Time {
+	err := s.loadData()
+	if err != nil {
+		return time.Time{}
+	}
+
+	return s.data.LastUpdate
+}
+
+func (s *fileStorage) Pull() (messages types.Manifest, err error) {
```

Comment:
Since Pull() and Push() have such established usages in DDEV it may be better to use a synonym here for what you're trying to accomplish. I can't make a suggestion yet because I'm not sure what Pull and Push mean yet.

Reply:
Pull and Push pulls and pushes messages from/to the storage. Terms are imho fine and widely used.